internal/plugin/grpc_client_plugin_test.go

```diff
@@ -61,20 +122,42 @@ func TestGrpcClient_NewGrpcClient(t *testing.T) {
 	}
 }
 
-func TestGrpcClient_InitWithInvalidServerAddr(t *testing.T) {
-	ctx := context.Background()
-	agentConfig := types.GetAgentConfig()
-	agentConfig.Command.Server.Host = "saasdkldsj"
+func TestGrpcClient_Init(t *testing.T) {
+	tests := []struct {
+		name          string
+		agentConfig   *config.Config
+		server        string
+		expectedError bool
+		errorMessage  string
+	}{
+		{
+			"Test 1: GRPC type specified in config",
+			types.GetAgentConfig(),
+			"incorrect-server",
+			true,
+			"connection error",
+		},
+	}
 
-	client := NewGrpcClient(agentConfig)
-	assert.NotNil(t, client)
+	for _, tt := range tests {
```

Comment:
Slightly confusing to call this `tt` seeing as the type for `t` and `ttt` are different.

Reply:
renamed in next commit

Reply:
there was a lot of this so refactored all the references I found